src/functions/vacation-request/update-vacation-request/handler.ts

```diff
@@ -0,0 +1,71 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { vacationRequestService } from "src/database/services";
+import type VacationRequestModel from "src/database/models/vacationRequest";
+import type vacationRequestSchema from "src/schema/vacationRequest";
+
+/**
+ * Lambda function to update a vacation request
+ *
+ * @param event event
+ */
+const updateVacationRequestHandler: ValidatedEventAPIGatewayProxyEvent<typeof vacationRequestSchema> = async event => {
+  const { pathParameters, body } = event;
+  const id = pathParameters?.id;
+  const {
+    personId,
+    draft,
+    startDate,
+    endDate,
+    days,
+    type,
+    message,
+    createdBy,
+    createdAt,
+    updatedAt
+  } = body;
+
+  if (!id) {
+    return {
+      statusCode: 400,
+      body: "Bad request, missing id"
+    };
+  }
+
+  const existingVacationRequest = await vacationRequestService.findVacationRequest(id);
+  if (!existingVacationRequest) {
+    return {
+      statusCode: 404,
+      body: `Vacation request ${id} not found`
+    };
+  }
+
+  const vacationRequestUpdates: VacationRequestModel = {
+    id: existingVacationRequest.id,
+    personId: personId,
+    draft: draft,
+    startDate: startDate,
+    endDate: endDate,
+    days: days,
+    type: type,
+    message: message,
+    createdBy: createdBy,
+    createdAt: createdAt,
```

Comment:
If we are updating perhaps we should not be changing the createdBy or CreatedAt. Does this mean we need an updated by?